webui/react/src/pages/ExperimentDetails/ExperimentActions.tsx

```diff
@@ -29,91 +29,68 @@ interface Props {
   onSettled: () => void; // A callback to trigger after an action is done.
 }
 
-type ButtonLoadingStates = Record<Action, boolean>;
-
 const ExperimentActions: React.FC<Props> = ({ experiment, onClick, onSettled }: Props) => {
-  const [ btnLoadingStates, setBtnLoadingStates ] = useState<ButtonLoadingStates>({
-    Activate: false,
-    Archive: false,
-    Cancel: false,
-    Fork: false,
-    Kill: false,
-    Pause: false,
-    Tensorboard: false,
-    Unarchive: false,
-  });
+  const [ isRunningActivate, setIsRunningActivate ] = useState<boolean>(false);
+  const [ isRunningArchive, setIsRunningArchive ] = useState<boolean>(false);
+  const [ isRunningCancel, setIsRunningCancel ] = useState<boolean>(false);
+  const [ isRunningKill, setIsRunningKill ] = useState<boolean>(false);
+  const [ isRunningPause, setIsRunningPause ] = useState<boolean>(false);
+  const [ isRunningUnarchive, setIsRunningUnarchive ] = useState<boolean>(false);
+
+  useEffect(() => {
+    setIsRunningArchive(false);
+    setIsRunningUnarchive(false);
+  }, [ experiment.archived ]);
+
+  useEffect(() => {
+    setIsRunningActivate(false);
+    setIsRunningCancel(false);
+    setIsRunningKill(false);
+    setIsRunningPause(false);
+  }, [ experiment.state ]);
 
   const handleArchive = useCallback(() => async (): Promise<void> => {
-    setBtnLoadingStates(state => ({ ...state, Archive: true }));
-    try {
-      await archiveExperiment({ experimentId: experiment.id });
-      onSettled();
-    } finally {
-      setBtnLoadingStates(state => ({ ...state, Archive: false }));
```

Comment:
like this but maybe just catch instead of finally?

Reply:
Right, that's better. Should be good now?